src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -367,11 +368,24 @@ public override SyntaxNode AsPublicInterfaceImplementation(SyntaxNode declaratio
             });
         }
 
+        public interface Foo
+        {
+            void Moo<T>() where T : IEnumerable<T>;
+        }
+
+        public class Bar : Foo
+        {
+            public void Moo<T>() where T : IEnumerable<T>
+            {
+            }
+        }
+
         public override SyntaxNode AsPrivateInterfaceImplementation(SyntaxNode declaration, SyntaxNode interfaceTypeName, string interfaceMemberName)
```

Comment:
Why the term 'Private' here?  I thought this was generally called an 'ExplicitInterfaceImplementation'.  That's also how we present this concept through features (i.e. "implement this interface 'explicitly'").